src/markdown-pages/install-with-kurl/migrating-csi.md

```diff
@@ -3,73 +3,206 @@ path: "/docs/install-with-kurl/migrating-csi"
-kind: Installer
-metadata:
-  name: new
-spec:
-  kubernetes:
-    version: 1.19.12
-  docker:
-    version: 20.10.5
-  weave:
-    version: 2.6.5
-  openebs:
-    version: 3.3.0
-    isLocalPVEnabled: true
-```
+| From      | To        | Notes                                                                                                                                                                                                                  |
+|-----------|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Longhorn  | OpenEBS 3.3.0 and later  | Migrating from Longhorn to OpenEBS 3.3.0 or later is recommended for single-node installations. |
+| Longhorn  | Rook 1.10.8 and later | Migrating from Longhorn to Rook 1.10.8 or later is recommended for clusters with three or more nodes where data replication and availability are requirements. Compared to OpenEBS, Rook requires more resources from your cluster, including a dedicated block device. Single-node installations of Rook are not recommended. Migrating from Longhorn to Rook is not supported for single-node clusters.  |
+| Rook      | OpenEBS 3.3.0 and later  | Migrating from Rook to OpenEBS 3.3.0 or later is strongly recommended for single-node installations, or for applications that do not require data replication. Compared to Rook, OpenEBS requires significantly fewer hardware resources from your cluster. |
+
+For more information about how to choose between Rook or OpenEBS, see [Choosing a PV Provisioner](create-installer/choosing-a-pv-provisioner).
+
+## About Changing the CSI Add-on
+
+You can change the CSI provisioner that your kURL cluster uses by updating the CSI add-on in your kURL specification file. Then, when you upgrade a kURL cluster using the new specification, the kURL installer detects the change that you made to the CSI add-on and begins automatically migrating data from the current provisioner to the new one.
+
+kURL supports data migration when you change your CSI provider from Rook to OpenEBS, or when you change from Longhorn to Rook or OpenEBS.
+For more information about how kURL automatically migrates data in either scenario, see the following sections:
+* [Rook to OpenEBS](#rook-to-openebs)
+* [Longhorn to Rook or OpenEBS](#longhorn-to-rook-or-openebs)
+
+### Rook to OpenEBS
+
+The following describes the automatic data migration process when you change the CSI provisioner add-on from Rook to OpenEBS:
+
+1. kURL recreates all PVCs that were originally created using Rook onto OpenEBS with the same name and contents. 
+
+1. If you are migrating off Rook from a cluster that has more than two nodes, OpenEBS attempts to create local volumes on the same nodes where the original Rook PVCs were referenced.
+
+1. kURL uninstalls Rook from the cluster.
+
+### Longhorn to Rook or OpenEBS
```

Comment:
I wasn't totally sure if it would make most sense to combine the Rook --> OpenEBS & Longhorn --> Rook/OpenEBS workflows into one, or keep them separate.

The content previously said that "the same applies" for Longhorn, though the Longhorn steps seemed to have pretty different detail, so I decided to keep them separate for this draft. Can edit that as needed

Reply:
Thanks @paigecalvert . I have changed a little bit the steps presented in the doc but it was overall correct.

Reply:
I guess that the original documentation was misguiding. The migration procedure is the same, regardless of source or destination provisioners. I have updated the doc to reflect that (in a section called `Migration process overview`) and removed those two separated sections. Can you please take a look if the phrasing is adequate ?

Reply:
Yes will do!

Reply:
Pushed a couple minor edits:
- Since the edits remove the need for two different subheadings, I figured we could condense this under the "About Changing the CSI Add-on" heading, which only had a couple sentences under it previously anyway. Updated the TOC at the top accordingly
- Minor wording edits (e.g., added a missing comma)